### PR TITLE
Incoming external calls in CDRs

### PR DESCRIPTION
**Pull request by mmadariaga**
This PR fixes https://github.com/irontec/ivozprovider/issues/956

**Notice** that a new column has been added in exported call CSV files. This may break third party integrations

### CODE REVIEW

File: doc/sphinx/administration_portal/brand/calls/call_csv_schedulers.rst [hunk 1]
```diff
@@ -8,20 +8,40 @@ This section allows programming the automatic periodical creation of CSV reports
...
          In this kind of schedules, a notification template can be chosen. In remaining
          schedules, the notification template assigned to the specific client will be used.
 
-Apart from the fields above, everything described in :ref:`Invoice schedulers` applies here:
+When adding a new definition, these fields are shown:
+
+.. glossary::
+
+    Name
+        Name of the scheduled invoice
```
Comment (Kaian) [inline doc/sphinx/administration_portal/brand/calls/call_csv_schedulers.rst]: Wrong scheduler here

File: doc/sphinx/administration_portal/brand/calls/call_csv_schedulers.rst [hunk 2]
```diff
@@ -8,20 +8,40 @@ This section allows programming the automatic periodical creation of CSV reports
...
+
+    Call direction:
+        Which kind of calls should be included: Inbound, outbound or both.
+
+    Client
+        Which client calls should be included
+
+    Email
+        Send generated invoices via email. Empty if no automatic mail is wanted.
```
Comment (Kaian) [inline doc/sphinx/administration_portal/brand/calls/call_csv_schedulers.rst]: Send generated Call CSV reports via email

File: library/Ivoz/Provider/Domain/Service/BillableCall/CsvExporter.php [hunk 3]
```diff
@@ -48,13 +50,15 @@ public function __construct(
...
+     * @param string | null $direction
      * @return string
      */
     public function execute(
         \DateTime $inDate,
         \DateTime $outDate,
         CompanyInterface $company = null,
-        BrandInterface $brand = null
+        BrandInterface $brand = null,
+        $direction = 'outbound'
```
Comment (Kaian) [inline library/Ivoz/Provider/Domain/Service/BillableCall/CsvExporter.php]: Could we use BillableCallInterface::DIRECTION_OUTBOUND here?

File: scheme/app/DoctrineMigrations/Version20190410082735.php [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+{
+    /**
+     * @param Schema $schema
+     */
+    public function up(Schema $schema)
+    {
+        // this up() migration is auto-generated, please modify it to your needs
+        $this->abortIf($this->connection->getDatabasePlatform()->getName() !== 'mysql', 'Migration can only be executed safely on \'mysql\'.');
+
+        $this->addSql('ALTER TABLE BillableCalls ADD direction VARCHAR(255) DEFAULT \'outbound\' COMMENT \'[enum:inbound|outbound]\'');
```
Comment (Kaian) [inline scheme/app/DoctrineMigrations/Version20190410082735.php]: 255 seems a bit excessive for a 10 character field. Do not know if mysql optimizes this.

File: web/admin/application/languages/es_ES/es_ES.po [hunk 5]
```diff
@@ -863,11 +861,22 @@ msgstr "Externa"
 msgid "External Filter destination"
 msgstr "Destino llamada filtro"
 
+msgid "External call"
+msgid_plural "External calls"
+msgstr[0] "Llamada facturable"
+msgstr[1] "Llamadas facturables"
```
Comment (Kaian) [inline web/admin/application/languages/es_ES/es_ES.po]: Is this translation intended? Afaik, not all external calls can be part of an invocie

File: library/Ivoz/Provider/Domain/Service/BillableCall/CsvExporter.php (hunk 3 above)
Comment (mmadariaga) [reply library/Ivoz/Provider/Domain/Service/BillableCall/CsvExporter.php]: Varchar columns are variable-length strings, I would not worry about this

File: scheme/app/DoctrineMigrations/Version20190410082735.php (hunk 4 above)
Comment (mmadariaga) [reply scheme/app/DoctrineMigrations/Version20190410082735.php]: Varchar columns are variable-length strings. I would not worry about this

File: web/admin/application/languages/es_ES/es_ES.po [hunk 6]
```diff
@@ -872,11 +870,22 @@ msgstr "Externa"
...
+msgstr[0] "Llamada externa"
+msgstr[1] "Llamadas externas"
+
 msgid "External call filter"
 msgid_plural "External call filters"
 msgstr[0] "Filtro de entrada externo"
 msgstr[1] "Filtros de entrada externo"
 
+msgid "External inbound calls"
+msgstr "Llamada externa entrante"
```
Comment (Kaian) [inline web/admin/application/languages/es_ES/es_ES.po]: This should be plural

File: web/admin/application/languages/es_ES/es_ES.po [hunk 7]
```diff
@@ -872,11 +870,22 @@ msgstr "Externa"
...
+
+msgid "External outbound calls"
+msgstr "Llamada external saliente"
```
Comment (Kaian) [inline web/admin/application/languages/es_ES/es_ES.po]: This also seems plural

File: library/DataFixtures/ORM/ProviderBillableCalls.php [hunk 8]
```diff
@@ -36,6 +36,7 @@ public function load(ObjectManager $manager)
                 $this->setStartTime($startTime);
                 $this->setCaller('+34633646464');
                 $this->setCallee('+34633656565');
+                $this->setDirection('outbound');
```
Comment (Kaian) [inline library/DataFixtures/ORM/ProviderBillableCalls.php]: Maybe constant also here?

**Pull request merged**